tests/test_bootstrap.py

```diff
@@ -0,0 +1,1390 @@
+
+        This version of the test covers the case of an equilibrium
+        with a current profile rather than an iota profile.
+        """
+        ne0 = 4.0e20
+        T0 = 12.0e3
+        ne = PowerSeriesProfile(ne0 * np.array([1, -1]), modes=[0, 10])
+        Te = PowerSeriesProfile(T0 * np.array([1, -1]), modes=[0, 2])
+        Ti = Te
+
+        helicity = (1, 0)
+        B0 = 5.0  # Desired average |B|
+        LM_resolution = 8
+
+        current = PowerSeriesProfile([0, -1.2e7], modes=[0, 2], sym=False)
+
+        # Set initial condition:
+        Rmajor = 6.0
+        aminor = 2.0
+        NFP = 1
+        surface = FourierRZToroidalSurface(
+            R_lmn=np.array([Rmajor, aminor]),
+            modes_R=[[0, 0], [1, 0]],
+            Z_lmn=np.array([-aminor]),
+            modes_Z=[[-1, 0]],
+            NFP=NFP,
+        )
+
+        eq = Equilibrium(
+            surface=surface,
+            electron_density=ne,
+            electron_temperature=Te,
+            ion_temperature=Ti,
+            current=current,
+            Psi=B0 * np.pi * (aminor**2),
+            NFP=NFP,
+            L=LM_resolution,
+            M=LM_resolution,
+            N=0,
+            L_grid=2 * LM_resolution,
+            M_grid=2 * LM_resolution,
+            N_grid=0,
+            sym=True,
+        )
+        current_L = 16
+        eq.current.change_resolution(current_L)
+
+        eq.solve(
+            verbose=3,
+            ftol=1e-8,
+            constraints=get_fixed_boundary_constraints(kinetic=True, iota=False),
+            optimizer=Optimizer("lsq-exact"),
+            objective=ObjectiveFunction(objectives=ForceBalance()),
+        )
+
+        eq.save("test_bootstrap_consistency_current_initial.h5")
+        # eq = desc.io.load("test_bootstrap_consistency_current_initial.h5")
+
+        # Done establishing the initial condition. Now set up the optimization.
+
+        constraints = (
+            ForceBalance(),
+            FixBoundaryR(),
+            FixBoundaryZ(),
+            FixElectronDensity(),
+            FixElectronTemperature(),
+            FixIonTemperature(),
+            FixAtomicNumber(),
+            FixCurrent(indices=[0]),
+            FixPsi(),
+        )
+
+        # grid for bootstrap consistency objective:
+        grid = QuadratureGrid(
+            L=current_L * 2,
+            M=eq.M * 2,
+            N=eq.N * 2,
+            NFP=eq.NFP,
+        )
+        objective = ObjectiveFunction(
+            BootstrapRedlConsistency(
+                grid=grid,
+                helicity=helicity,
+            )
+        )
+        eq, _ = eq.optimize(
+            verbose=3,
+            objective=objective,
+            constraints=constraints,
+            optimizer=Optimizer("scipy-trf"),
+            ftol=1e-6,
+            gtol=0,  # It is critical to set gtol=0 when optimizing current profile!
+        )
+
+        eq.save("test_bootstrap_consistency_current_final.h5")
```

Comment:
same note about saving to temp directory

Reply:
Now `TmpDir` is used.